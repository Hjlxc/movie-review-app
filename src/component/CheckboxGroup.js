import React from 'react';
import { Checkbox } from 'antd';

import { VerticalFlexWrapper } from '../container/styledComponent';

export default function({ options, checked, onOptionClick }) {
  return (
    <VerticalFlexWrapper>
      {options.map((option, idx) => (
        <div key={option}>
          <Checkbox
            data-testid={`checkBox-${idx}`}
            onChange={() => onOptionClick(option, checked[option])}
            checked={checked[option]}
          >
            {option}
          </Checkbox>
        </div>
      ))}
    </VerticalFlexWrapper>
  );
}
