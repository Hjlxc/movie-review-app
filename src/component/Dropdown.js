import React from 'react';

import { Menu, Dropdown } from 'antd';
import { DownOutlined } from '@ant-design/icons';

export default function({ select, options, onChange }) {
  const menu = (
    <Menu onClick={onChange}>
      {options.map((option) => (
        <Menu.Item key={option}>{option}</Menu.Item>
      ))}
    </Menu>
  );

  return (
    <Dropdown overlay={menu}>
      <a
        className="ant-dropdown-link"
        href="/"
        onClick={(e) => e.preventDefault()}
      >
        {select} <DownOutlined />
      </a>
    </Dropdown>
  );
}
