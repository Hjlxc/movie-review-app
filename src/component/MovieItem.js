import React from 'react';

import { Rate } from 'antd';

import {
  MovieItemWrapper,
  MovieItemSwatch,
  BoldWrapper,
} from '../container/styledComponent';

export default function MovieItem({ swatch, title, rating, onClick }) {
  return (
    <MovieItemWrapper>
      <MovieItemSwatch src={swatch} onClick={onClick} alt={`${title} Poster`} />
      <BoldWrapper>{title}</BoldWrapper>
      <Rate allowHalf disabled value={rating} />
    </MovieItemWrapper>
  );
}
