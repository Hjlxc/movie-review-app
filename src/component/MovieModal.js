import React from 'react';
import { Modal, Progress, Tooltip } from 'antd';

import {
  HorizontalFlexWrapper,
  MovieModalSwatch,
  VerticalFlexWrapper,
  BoldWrapper,
  TextWrapper,
  MovieModalSwatchWrapper,
  Gap,
} from '../container/styledComponent';

export default function MovieModal({
  swatch,
  title,
  vote_average,
  vote_count,
  overview,
  release_date,
  original_language,
  visible,
  onClose,
}) {
  return (
    <Modal
      title={title}
      centered
      visible={visible}
      footer={null}
      onCancel={onClose}
    >
      <HorizontalFlexWrapper>
        <MovieModalSwatchWrapper>
          <MovieModalSwatch src={swatch} alt={`${title} Poster`} />
        </MovieModalSwatchWrapper>
        <VerticalFlexWrapper>
          <HorizontalFlexWrapper>
            <TextWrapper>{release_date}</TextWrapper>
            <Gap />
            <TextWrapper>{`Language: ${original_language}`}</TextWrapper>
          </HorizontalFlexWrapper>
          <div>
            <Tooltip placement="right" title={`${vote_count} Voting`}>
              <Progress
                width={40}
                type="circle"
                percent={vote_average * 10}
                strokeColor={getColorGradient(vote_average)}
              />
            </Tooltip>
          </div>
          <BoldWrapper>Overview</BoldWrapper>
          <TextWrapper>{overview}</TextWrapper>
        </VerticalFlexWrapper>
      </HorizontalFlexWrapper>
    </Modal>
  );
}

// get color gradient based on rating
// 0-10 -> red-green
export const getColorGradient = (rating) => {
  const ratio = rating / 10;
  return `rgb(${255 - 255 * ratio},${255 * ratio},0)`;
};
