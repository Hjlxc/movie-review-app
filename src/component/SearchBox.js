import React from 'react';
import { Input } from 'antd';

export default function({ placeholder, onSearch }) {
  return (
    <Input.Search placeholder={placeholder} onSearch={onSearch} enterButton />
  );
}
