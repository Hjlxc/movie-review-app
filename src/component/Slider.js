import React from 'react';
import { Slider } from 'antd';

import { VerticalFlexWrapper, BoldWrapper } from '../container/styledComponent';

export default function({ title, min, max, selectMin, selectMax, onChange }) {
  const marks = { [min]: min, [max]: max };
  return (
    <VerticalFlexWrapper>
      {title && <BoldWrapper>{title}</BoldWrapper>}
      <Slider
        min={min}
        max={max}
        range
        defaultValue={[selectMin, selectMax]}
        step={0.1}
        onChange={onChange}
        marks={marks}
      />
    </VerticalFlexWrapper>
  );
}
