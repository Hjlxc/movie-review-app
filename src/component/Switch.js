import React from 'react';
import { Switch } from 'antd';

export default function({ checked, onChange }) {
  return <Switch checked={checked} onChange={onChange} />;
}
