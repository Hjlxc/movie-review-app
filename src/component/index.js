import CheckboxGroup from './CheckboxGroup';
import Slider from './Slider';
import MovieItem from './MovieItem';
import MovieModal from './MovieModal';
import Switch from './Switch';
import SearchBox from './SearchBox';

export { CheckboxGroup, Slider, MovieItem, MovieModal, Switch, SearchBox };
