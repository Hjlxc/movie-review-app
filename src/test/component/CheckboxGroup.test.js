import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from '@testing-library/react';

import { CheckboxGroup } from '../../component';

const options = ['a', 'b', 'c'];
const checked = { b: true, c: true };

describe('Test Switch Component', () => {
  let testRenderer;
  let instance;
  const mockOnChange = jest.fn((check) => {});
  beforeAll(() => {
    testRenderer = renderer.create(
      <CheckboxGroup
        options={options}
        checked={checked}
        onOptionClick={mockOnChange}
      />
    );
    instance = testRenderer.root;
  });

  it('Test switch render with correct props', () => {
    expect(instance.props.options).toEqual(options);
    expect(instance.props.checked).toEqual(checked);

    // check component render correct checkbox
    const checkBoxInputs = instance.findAllByType('input');
    expect(checkBoxInputs.length).toBe(options.length);

    // check each check box has correct props
    options.forEach((option, idx) => {
      const input = checkBoxInputs[idx];
      expect(input.props.checked).toBe(!!checked[option]);
    });
  });

  it('matches snapshot', () => {
    const tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test user interactive', () => {
  const mockOnChange = jest.fn();
  beforeEach(() => {
    cleanup();
    mockOnChange.mockClear();
  });

  it('test user click on checkbox', () => {
    const { getByTestId } = render(
      <CheckboxGroup
        options={options}
        checked={checked}
        onOptionClick={mockOnChange}
      />
    );
    options.forEach((option, idx) => {
      fireEvent.click(getByTestId(`checkBox-${idx}`));
      expect(mockOnChange).toHaveBeenCalledTimes(idx + 1);
      expect(mockOnChange).toBeCalledWith(option, checked[option]);
    });
  });
});
