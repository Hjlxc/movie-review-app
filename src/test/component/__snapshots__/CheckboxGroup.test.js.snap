// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Switch Component matches snapshot 1`] = `
<div
  className="sc-AxiKw jWoTjM"
>
  <div>
    <label
      className="ant-checkbox-wrapper"
    >
      <span
        className="ant-checkbox"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-checkbox-input"
          data-testid="checkBox-0"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <span
          className="ant-checkbox-inner"
        />
      </span>
      <span>
        a
      </span>
    </label>
  </div>
  <div>
    <label
      className="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
    >
      <span
        className="ant-checkbox ant-checkbox-checked"
        style={Object {}}
      >
        <input
          checked={true}
          className="ant-checkbox-input"
          data-testid="checkBox-1"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <span
          className="ant-checkbox-inner"
        />
      </span>
      <span>
        b
      </span>
    </label>
  </div>
  <div>
    <label
      className="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
    >
      <span
        className="ant-checkbox ant-checkbox-checked"
        style={Object {}}
      >
        <input
          checked={true}
          className="ant-checkbox-input"
          data-testid="checkBox-2"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <span
          className="ant-checkbox-inner"
        />
      </span>
      <span>
        c
      </span>
    </label>
  </div>
</div>
`;
